msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java

```diff
@@ -202,7 +202,7 @@ public JsonToken nextToken()
             case STRING:
                 value = messageUnpacker.unpackValue(var);
                 if (parsingContext.inObject() && _currToken != JsonToken.FIELD_NAME) {
-                    parsingContext.setCurrentName(value.asRawValue().stringValue());
+                    parsingContext.setCurrentName(value.asRawValue().toJson());
```

Comment:
I'm surprised this works. I would have thought that the `toJson` would have quoted the string (as `ImmutableRawValueImpl` does). You actually get a `Variable$StringValueAccessor` which does't quote the string.

This depends on the output format of `toJson` however you wrote in the [java doc][1] 

> Do not write code that depends on the resulting json format.

but this depends directly on the output. Should this be using `toJson` or a method that would be more predictable?

   [1]: https://github.com/msgpack/msgpack-java/pull/265/files#diff-1f56e7353af89c0830ab0ade1552c7e2R253